bgpd/bgp_vty.c

```diff
@@ -1673,6 +1673,31 @@ DEFUN (no_bgp_graceful_restart_restart_time,
   return CMD_SUCCESS;
 }
 
+DEFUN (bgp_graceful_restart_preserve_fw,
+       bgp_graceful_restart_preserve_fw_cmd,
+       "bgp graceful-restart preserve-fw-state",
```

Comment:
Instead of creating two new DEFUNs you could make "preserve-fw-state" an optional keyword at the end of the existing "bgp graceful-restart" DEFUN.  It's minor though...up to you.